sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -689,6 +689,11 @@ case class DescribeColumnCommand(
       buffer += Row("distinct_count", cs.map(_.distinctCount.toString).getOrElse("NULL"))
       buffer += Row("avg_col_len", cs.map(_.avgLen.toString).getOrElse("NULL"))
       buffer += Row("max_col_len", cs.map(_.maxLen.toString).getOrElse("NULL"))
+      buffer ++= cs.flatMap(_.histogram.map { hist =>
+        val header = Row("histogram", s"height: ${hist.height}, num_of_bins: ${hist.bins.length}")
+        Seq(header) ++ hist.bins.map(bin =>
+          Row("", s"lower_bound: ${bin.lo}, upper_bound: ${bin.hi}, distinct_count: ${bin.ndv}"))
```

Comment:
add a comment that the empty column is for indentation and better readability?

Reply:
@wzhfy I'd rather define a `val` with the comment being the name of the val. That would make it "compile-safe".

Reply:
@mgaido91 @jaceklaskowski How about using "histogram_bin" instead of the empty string?
```
histogram	height: 2.0, num_of_bins: 2
histogram_bin	lower_bound: 1.0, upper_bound: 2.0, distinct_count: 2
histogram_bin	lower_bound: 2.0, upper_bound: 4.0, distinct_count: 2
```

Reply:
And how about making each class member as a separate field?
```
histogram	height: 2.0	num_of_bins: 2
histogram_bin	lower_bound: 1.0	upper_bound: 2.0	distinct_count: 2
histogram_bin	lower_bound: 2.0	upper_bound: 4.0	distinct_count: 2
```

Reply:
maybe adding the bin id?
```
histogram	height: 2.0, num_of_bins: 2
bin0      lower_bound: 1.0, upper_bound: 2.0, distinct_count: 2
bin1      lower_bound: 2.0, upper_bound: 4.0, distinct_count: 2
```

Reply:
Do you think this (printing one line per bucket) might become too verbose if there are many buckets? In an interactive session writing 256 buckets, one-per-line is likely to push the non-histogram output off the top of the screen.

Reply:
kind of, but I don't have a better idea, adding a new flag to `DESC COLUMNS` seems an overkill.

Reply:
yeah, I was wondering if we actually need to see every individual bucket instead of a summary (max bucket: 2.0->4.0 w/128 entries, min bucket: 6.0->8.0 w/0 entries, median bucket size: 32).

Reply:
for debug purpose, I'd like to have as more information as possible. In the future maybe we can simplify the `DESC COLUMN` and use `spark.catalog.getColumn` to get detailed information.

Reply:
What is the output contents/formats of the other vendors?

Reply:
@gatorsmile In Hive, there isn't yet a histogram implementation (HIVE-3526). In Oracle and MySQL, the information is stored in metadata tables which can be queried (https://docs.oracle.com/cloud/latest/db112/REFRN/statviews_2106.htm#REFRN20279) and here we have mainly two information:
 - the cumulative count so far;
 - the endpoint value for the current bin.

Reply:
@henryr I think verbosity is a minor issue, since we are returning a `DataFrame` and the user can decide what to show.